chcemvediet/apps/anonymization/cron.py

```diff
@@ -25,6 +33,47 @@ def attachment_normalization():
             content_type=content_types.PDF_CONTENT_TYPE
         )
         cron_logger.info(u'Normalized attachment: {}'.format(attachment))
+
+    elif attachment.content_type in content_types.LIBREOFFICE_CONTENT_TYPES:
+        directory_name = tempfile.mkdtemp(dir=u'.')
+        file_copy = os.path.join(directory_name, u'file' + guess_extension(attachment.content_type))
+        shutil.copy2(attachment.file.path, file_copy)
+
+        p = subprocess32.Popen(
+            [u'libreoffice', u'--headless', u'--convert-to', u'pdf', u'--outdir', directory_name,
+             file_copy],
+            stdout=subprocess32.PIPE,
+            stderr=subprocess32.PIPE
+        )
+        try:
+            out, err = p.communicate(timeout=300)
+        except subprocess32.TimeoutExpired:
+            p.terminate()
+            out, err = p.communicate()
+            err += u'Timeout expired.'
+
+        if u'file.pdf' in os.listdir(directory_name):
+            file_pdf = open(os.path.join(directory_name, u'file.pdf'), u'rb')
+            AttachmentNormalization.objects.create(
+                attachment=attachment,
+                successful=True,
+                file=ContentFile(file_pdf.read()),
+                content_type=content_types.PDF_CONTENT_TYPE,
+                debug=u'{}\n{}'.format(out, err)
+            )
+            cron_logger.info(u'Normalized attachment using libreoffice: {}'.format(attachment))
+        else:
+            AttachmentNormalization.objects.create(
+                attachment=attachment,
+                successful=False,
```

Comment:
Tu pozor, dovodov preco konverzia failne moze byt v skutocnosti ovela viac. Co ak failne `open(os.path.join(directory_name, u'file.pdf'), u'rb')` v predoslej vetve?

Ovela praktickejsie je "uspesnu" vetvu zabalit do `try`/`except` a `successful=False` savnut do DB ked nieco v `try`/`except` raisne exception. Potom vlastne ani nemusis explicitne kontrolovat `if u'file.pdf' in os.listdir(directory_name)`, len staci skusit ten subor otvorit a ak neexistuje, tak to raisne exception, ktoru v `catch` zachytis a savnes neuspesny pokus.

Ked do `try`/`except` zabalis aj cely `subprocess.run()` tak ani nepotrebujes specialne ohandlovavat `except subprocess32.TimeoutExpired` ale proste odchytis vsetky mozne exceptions.

V pripade exception, do `debug` zapis aj zachytenu exception.

Reply:
;ok: